### PR TITLE
Support Auth on LB with OIDC compliant provider

### PR DESCRIPTION
**Pull request by jonnyshaw89**
This PR allows the LoadBalancer to check the Authentication of a user before forwarding the request to a container.

### CODE REVIEW

File: ecs.tf [hunk 1]
```diff
@@ -32,19 +32,35 @@ resource "aws_alb_listener_rule" "listener_rule" {
...
+  action {
+    type = "authenticate-oidc"
+
+    authenticate_oidc {
+      authorization_endpoint = "${var.auth_domain}authorize"
+      client_id              = "${var.auth_client_id}"
+      client_secret          = "${var.auth_client_secret}"
+      issuer                 = "${var.auth_domain}"
+      token_endpoint         = "${var.auth_domain}oauth/token"
+      user_info_endpoint     = "${var.auth_domain}userinfo"
```
Comment (samiwel) [inline ecs.tf]: Maybe make the entire string a variable?

File: ecs.tf [hunk 2]
```diff
@@ -32,19 +32,35 @@ resource "aws_alb_listener_rule" "listener_rule" {
...
+      on_unauthenticated_request = "${var.auth_unauth_action}"
+      scope = "openid profile email"
```
Comment (samiwel) [inline ecs.tf]: Should the scopes also be configurable?

**Review by samiwel — CHANGES_REQUESTED** [review]
Looks good, just a couple of observations.

File: ecs.tf (hunk 1 above)
Comment (tgandrews) [reply ecs.tf]: Would you do it for the `token_endpoint` and `authorization_endpoint` above as well? Or do you feel those are generic and `user_info` isn't

**Review by tgandrews — APPROVED** [review]
As much as I understand this makes sense.

**Comment by jonnyshaw89** [comment]
@samiwel ready for a re-review

**Review by samiwel — APPROVED** [review]
Changes look good to me.

**Pull request merged**